File: common/src/main/java/io/github/notenoughupdates/moulconfig/annotations/ConfigOverride.java
package io.github.notenoughupdates.moulconfig.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.FIELD)
public @interface ConfigOverride {
    /**
     * Controls placement of this field relative to its siblings, inheriting the overridden parent
     * field's {@link ConfigOrder} value by default. Set explicitly to override that behavior.
     * Uses {@link Integer#MIN_VALUE} as a sentinel to indicate inheritance. Do not use that value directly.
     */
    int overrideOrder() default Integer.MIN_VALUE;
}

File: common/src/main/java/io/github/notenoughupdates/moulconfig/annotations/ConfigOverride.kt
package io.github.notenoughupdates.moulconfig.annotations

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FIELD)
annotation class ConfigOverride(
    /**
     * Marks this field as intentionally overriding a parent class field of the same name.
     * Optionally specify an order to control placement relative to siblings; lower values appear first.
     */
    val order: Int = 0
)
